### PR TITLE
Correct eos_token_id settings in generate

### PR DESCRIPTION
**Pull request by thinksoso**
In generation process, if the eos_token_id and config.eos_token_id are both None, use config.decoder.eos_token_id to set eos_token_id

Fixes # (issue)
https://github.com/huggingface/transformers/issues/14905

### CODE REVIEW

File: src/transformers/generation_utils.py [hunk 1]
```diff
@@ -1049,6 +1049,8 @@ def generate(
 
         pad_token_id = pad_token_id if pad_token_id is not None else self.config.pad_token_id
         eos_token_id = eos_token_id if eos_token_id is not None else self.config.eos_token_id
+        if eos_token_id is None and hasattr(self.config, "decoder"):
```
Comment (ngoquanghuy99) [inline src/transformers/generation_utils.py:1052]: Could you give code to reproduce or any case of this?

**Review by ngoquanghuy99 — APPROVED** [review]
I think it's fine! 
What do you think? @patrickvonplaten

**Comment by patrickvonplaten** [comment]
Good for me as well!

**Comment by patrickvonplaten** [comment]
@thinksoso, could you try to adapt the failing test by setting `config.decoder.eos_token_id = None` before running it? :-)

**Comment by thinksoso** [comment]
> @thinksoso, could you try to adapt the failing test by setting `config.decoder.eos_token_id = None` before running it? :-)

I made it! But the ci failed, I think it is not caused by my changes.

**Comment by thinksoso** [comment]
> I think it's fine! What do you think? @patrickvonplaten

Sorry, I do the rebase, and the commits explode. Please only focus my own two commits. Thanks!

**Review by patrickvonplaten — APPROVED** [review]

**Pull request merged**